src/components/Button/Button.tsx

```diff
@@ -179,6 +194,14 @@ class Button extends UIComponent<Extendable<IButtonProps>, any> {
       onClick(e, this.props)
     }
   }
+
+  private handleFocus = (e: React.SyntheticEvent) => {
+    const { onFocus } = this.props
+    this.setState({ isLastFocusFromKeyboard: whatInput.ask() === 'keyboard' })
+    if (onFocus) {
+      onFocus(e, this.props)
```

Comment:
Shouldn't we use _.invoke, like this?
```_.invoke(this.props, 'onFocus', e, this.props)```
I see in some places we call it directly in others using ```_.invoke```, shouldn't we be consistent or is there any reason for using it here like this? Probably it's more question to @kuzhelov @mnajdova

Reply:
my vote would be to use just any safe approach that would work - in other words, as long as it is safe and declarative enough (so it is easy to read), I would be fine with any approach chosen. 

For this particular example, frankly, doesn't see much value in using `invoke` instead - but this is just my opinion.

The only thing I might suggest is to simplify this expression a bit:

```js
  ...
  onFocus && onFocus(e, this.props)
```

Reply:
I prefer this invoking of the functions as well, as they look quite more simple and direct about what we want to do.

Reply:
I vote for consistency = `_.invoke()`